daemon/src/main.rs

```diff
@@ -752,6 +758,19 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         let wayland_event = !fds[0].revents().is_empty();
         let socket_event = !fds[1].revents().is_empty();
 
+        if socket_event {
```

Comment:
This socket event cannot be moved up here or it will lead to ordering problems. Instead of putting `if daemon.paused { continue }`, you add this condition to the `daemon.draw()` at the end, so it becomes `if daemon.poll_time.is_some() && !daemon.paused { daemon.draw() }`

Reply:
Ah I see, I'll fix that